### ISSUE TITLE
Object detection for robotic picking?

### ISSUE THREAD

**Issue by robotwhispering** [AlexeyAB/darknet#4859]
@AlexeyAB can YOLO or other detection algorithm provide accurate enough object detection of custom object so a robot can use the coordinates to pick from? 

The idea is to train a custom dataset of an object that doesn't exist in any of the common pre-trained weights files. Example is a dataset of bolts. The bolts might be different widths and lengths, but would have similar characteristics such as threads, hex heads, etc. Each image of a bolt would be oriented such that the bolt is standing up straight, perfectly vertical, and the bolts I want to pick with the robot would be in the same orientation. The only axis that would vary would be Z-axis because the height of bolts to pick will vary, but their position on the conveyor would always be the same.  I assume to draw boundary boxes from the top edge of the bolt to the bottom edge of the bolt.

Do I need to train the weights with EVERY possible bolt length that might be picked? Or can YOLO learn enough about what bolts look like to automatically adjust the boundary box based on how tall the box is?

**Comment by AlexeyAB** [comment]
> can YOLO or other detection algorithm provide accurate enough object detection of custom object so a robot can use the coordinates to pick from?

Yes, it can in your case.

> Do I need to train the weights with EVERY possible bolt length that might be picked?

Yes, you should.

**Comment by robotwhispering** [comment]
@AlexeyAB thank you for your quick response. The difficulty is that the bolts on the conveyor, some of them are new bolts of a specific length, and I can train lots of images of those. But some of the bolts have been grinded down to shorter, unknown length. So there could technically be an infinite number is unknown lengths.

**Comment by AlexeyAB** [comment]
Data augmentation (jitter) will solve this issue, by randomly changing size and aspect ratio of images. But the more images with every possible bolt length in training dataset - the better result.

There should be the smallest and largest bolts in training set. Yolo will generalize and interpolate other lengths.
